ddtrace/llmobs/_utils.py

```diff
@@ -41,53 +47,125 @@
+    if strict_validation:
+        _strict_validate_prompt(prompt)
+
+    # Ensure only one of `template` or `chat_template` is provided
+    if template and chat_template:
+        raise ValueError("Only one of 'template' or 'chat_template' can be provided, not both.")
+
+    # Stage 3: Set defaults
+    final_ml_app = ml_app or DEFAULT_PROMPT_ML_APP
+    final_prompt_id = prompt_id or f"{final_ml_app}_{DEFAULT_PROMPT_NAME}"
+    final_ctx_variable_keys = ctx_variable_keys or ["context"]
+    final_query_variable_keys = query_variable_keys or ["question"]
+
+    # Stage 4: Type checks
+    if not isinstance(final_prompt_id, str):
+        raise TypeError(f"'id' must be str, got {type(final_prompt_id).__name__}.")
+
+    if not (isinstance(final_ctx_variable_keys, list) and all(isinstance(i, str) for i in final_ctx_variable_keys)):
+        raise TypeError("'rag_context_variables' must be a List[str].")
+
+    if not (isinstance(final_query_variable_keys, list) and all(isinstance(i, str) for i in final_query_variable_keys)):
+        raise TypeError("'rag_query_variables' must be a List[str].")
+
+    if version and not isinstance(version, str):
+        raise TypeError(f"'version' must be str, got {type(version).__name__}.")
+
+    if tags:
+        if not isinstance(tags, dict):
+            raise TypeError(f"'tags' must be Dict, got {type(tags).__name__}")
+        if not all(isinstance(k, str) for k in tags):
+            raise TypeError("Keys of 'tags' must all be strings.")
+        if not all(isinstance(k, str) for k in tags.values()):
+            raise TypeError("Values of 'tags' must all be strings.")
+
+    if template and not isinstance(template, str):
+        raise TypeError(f"'template' must be str, got {type(template).__name__}.")
+
+    if chat_template:
+        if not isinstance(chat_template, list):
+            raise TypeError("'chat_template' must be a list.")
+        for ct in chat_template:
+            if not (isinstance(ct, dict) and {"role", "content"}.issubset(ct)):
+                raise TypeError("Each 'chat_template' entry should be dict[str,str] with role and content keys.")
+
+    if variables:
         if not isinstance(variables, dict):
-            raise TypeError("Prompt variables must be a dictionary.")
-        if not any(isinstance(k, str) or isinstance(v, str) for k, v in variables.items()):
-            raise TypeError("Prompt variable keys and values must be strings.")
+            raise TypeError(f"'variables' must be Dict, got {type(variables).__name__}")
+        if not all(isinstance(k, str) for k in variables):
+            raise TypeError("Keys of 'variables' must all be strings.")
+
+    # Stage 5: Transformations
+    # Ensure chat_template is standardized List[dict[role:str, content:str]]
+    final_chat_template = None
+    if chat_template:
+        final_chat_template = []
+        for msg in chat_template:
+            if isinstance(msg, dict):
```

Comment:
Do we need this check if we've already done the type checking in the previous stage?